### ISSUE TITLE
Python and .NET nightly tests failing with timeout

### ISSUE THREAD

**Issue by lblackstone** [pulumi/pulumi-eks#573]
Every nightly CI test for Python and .NET has failed starting on April 28th. This doesn't appear to be related to a change in the EKS repo, which last had changes merged on April 19th.

The error messages are not informative, and the reported failure is simply that the tests timed out after three hours.

**Comment by lblackstone** [comment]
I opened https://github.com/pulumi/pulumi-eks/pull/572 in an effort to get more debugging information, but the Python and .NET tests failed on the PR as well.

**Comment by justinvp** [comment]
Running `pulumi up -d` for the following program locally:

```python
import pulumi
import pulumi_eks as eks

cluster1 = eks.Cluster(f"mycluster")

pulumi.export("kubeconfig1", cluster1.kubeconfig)
```

with `pulumi_kubernetes==3.1.1`, the preview hangs and I see the following in the debug logs:

```
    debug: RPC failed with exception:
    debug: Traceback (most recent call last):
      File "/Users/justin/Desktop/2021/05/10/ekstest/venv/lib/python3.9/site-packages/pulumi/runtime/rpc_manager.py", line 67, in rpc_wrapper
        result = await rpc
      File "/Users/justin/Desktop/2021/05/10/ekstest/venv/lib/python3.9/site-packages/pulumi/runtime/resource.py", line 542, in do_register
        rpc.resolve_outputs(res, resolver.serialized_props, resp.object, deps, resolvers, transform_using_type_metadata)
      File "/Users/justin/Desktop/2021/05/10/ekstest/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 697, in resolve_outputs
        translated_value = translate_output_properties(value, translate_to_pass, types.get(key),
      File "/Users/justin/Desktop/2021/05/10/ekstest/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 614, in translate_output_properties
        types = _types.output_type_types(typ)
      File "/Users/justin/Desktop/2021/05/10/ekstest/venv/lib/python3.9/site-packages/pulumi/_types.py", line 841, in output_type_types
        return _types_from_py_properties(output_type_cls)
      File "/Users/justin/Desktop/2021/05/10/ekstest/venv/lib/python3.9/site-packages/pulumi/_types.py", line 736, in _types_from_py_properties
        cls_hints = get_type_hints(prop.fget, globalns=globalns, localns=localns)
      File "/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/typing.py", line 1386, in get_type_hints
        value = _eval_type(value, globalns, localns)
      File "/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/typing.py", line 256, in _eval_type
        ev_args = tuple(_eval_type(a, globalns, localns, recursive_guard) for a in t.__args__)
      File "/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/typing.py", line 256, in <genexpr>
        ev_args = tuple(_eval_type(a, globalns, localns, recursive_guard) for a in t.__args__)
      File "/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/typing.py", line 254, in _eval_type
        return t._evaluate(globalns, localns, recursive_guard)
      File "/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/typing.py", line 493, in _evaluate
        eval(self.__forward_code__, globalns, localns),
      File "<string>", line 1, in <module>
      File "/usr/local/Cellar/python@3.9/3.9.0_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/util.py", line 250, in __getattribute__
        raise ValueError(f"module object for {original_name!r} "
``` 

If I install `pulumi_kubernetes==3.1.0` and run `pulumi preview`, the preview completes successfully.

`pulumi_kubernetes` 3.1.1 introduced lazy loading of sub modules:

https://github.com/pulumi/pulumi-kubernetes/blob/19794052d49eb5b014ee7f96307bfd952db99b40/sdk/python/pulumi_kubernetes/core/__init__.py#L12

When we try to get the type hints for the following output property, we're running into that error:

https://github.com/pulumi/pulumi-eks/blob/4f4a75b17de98cf2f9c3d34a960b59503cbc4f0a/python/pulumi_eks/outputs.py#L628-L631

Two issues:
1. Any exceptions raised from `translate_output_properties` should not cause a preview/update to hang.
2. We need to be able to resolve the type when lazy module imports are used.

**Comment by lblackstone** [comment]
I built the `pulumi_kubernetes` package using the changes from https://github.com/pulumi/pulumi/pull/7024 and it seems to fix the Python preview hangs.

It seems like the .NET issue may be a separate problem.

**Comment by t0yv0** [comment]
Could you include a link to nightlies? I'd like to see when nightlies break I don't think I'm subscribed currently 🙏

**Comment by lblackstone** [comment]
It appears that #575 and the release of pulumi_kubernetes `v3.1.2` have [resolved the failing tests](https://github.com/pulumi/pulumi-eks/actions/runs/836394667).

We're not completely sure how #575 was related because the tracing endpoint had been removed some time ago. One theory is that is could have been related to DNS expiration.

These test failures also covered up the Python lazy loading regression that was fixed in the v3.1.2 release of the k8s provider.

**Comment by t0yv0** [comment]
Negative on DNS. I've tested our open tracing functions with busted addresses, they do not seem to block.
